app/services/bulk_invoice_service.rb

```diff
@@ -7,9 +7,8 @@ def initialize
 
   def start_pdf_job(order_ids)
     pdf = CombinePDF.new
-    orders = Spree::Order.where(id: order_ids)
 
-    orders.each do |order|
+    orders_from(order_ids).each do |order|
```

Comment:
This will load all orders into the memory at once and then loop through them. It would be better to limit memory consumption by loading orders in batches:
```suggestion
    orders_from(order_ids).find_each do |order|
```
https://apidock.com/rails/ActiveRecord/Batches/find_each

This is out of scope of this PR though. You don't have to do that change but if you do, it should be in a separate commit.

Reply:
@mkllnk Ok, I was not expecting users to check more than 2000 chekboxes but I may miss a usage case :) Batches is a security for sure.

Reply:
I'm not sure but I thought there was a checkbox for "check all". And if we don't have it yet, it may come in the future.

Reply:
I'd also like to find the orders in batches but `#find_each` clashes with the `order("completed_at DESC")`, as stated in https://apidock.com/rails/v3.2.13/ActiveRecord/Batches/find_in_batches

> It’s not possible to set the order. That is automatically set to ascending on the primary key (“id ASC”) to make the batch ordering work.